agent/fw_drupal8.c

```diff
@@ -322,12 +321,57 @@ NR_PHP_WRAPPER_END
  * function call of this type gets to name the txn.
  */
 NR_PHP_WRAPPER(nr_drupal8_name_the_wt_cached) {
-  const char* name = "page_cache";
+  char* name = NULL;
   zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;
+  zval* request = NULL;
+
+  zend_function* nrfn = NULL;
+
+  zval* controller = NULL;
 
   (void)wraprec;
 
   NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_DRUPAL8);
+
+  request = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS);
+
+  if (0 == nr_php_is_zval_valid_object(request)) {
+    nrl_verbosedebug(
+        NRL_TXN, "Drupal 8 PageCache::get does not have a request parameter");
+
+    goto end;
+  }
+
+  if (false
+      == nr_php_object_instanceof_class(
+          request, "Symfony\\Component\\HttpFoundation\\Request")) {
+    nrl_verbosedebug(
+        NRL_TXN, "Drupal 8 PageCache::get parameter is a non-Request object");
+    goto end;
+  }
+
+  if (!NRINI(drupal_page_cache_naming)) {
+    name = nr_strdup("page_cache");
+    goto end;
+  }
+
+  nrfn = nr_php_find_function("newrelic_name_cached");
+  if (NULL == nrfn) {
+    nrl_warning(NRL_FRAMEWORK,
+                "Failed to retrieve NR cached page naming function");
+    goto end;
```

Comment:
This could probably set `name` to "page_cache" since this is what we just did above when naming wasn't requested.  Maybe change the warning to include a notice that "page_cache" was being used as the name because of the failure.

Reply:
Maybe a better solution to multiple `name = nr_strdup("page_cache")` checks is to just check after `NR_PHP_WRAPPER_CALL` if `name == NULL` and set it to `page_cache` that way?

Reply:
My concern was that it appeared we had a case where we missed naming - if your solution handles these it should work fine.

Reply:
I think 04cbaf6d702fdb4aa2ff99624c3861a2870b0d25 is a slight improvement on the fallback naming

Reply:
On testing, that seems to have broken something... I'll have to look into this a little more